### PR TITLE
1. Added data type, matrix init and hot calls.

### PR DESCRIPTION
**Pull request by jkottiku**
2. Set matrix sizes properly in IET module.
3. iet workload thread functions definition changes.
4. rvs_blas interfaces function definition changes.
5. Remove redundant code.

### CODE REVIEW

File: iet.so/src/iet_worker.cpp [hunk 1]
```diff
@@ -114,73 +114,66 @@ IETWorker::IETWorker():endtest(false) {
...
     std::chrono::time_point<std::chrono::system_clock> iet_start_time, iet_end_time;
-    double timetakenforoneiteration;
-    double gflops_interval;
-    double duration;
-    uint64_t gem_ops;
+    double duration = 0;
     std::unique_ptr<rvs_blas> gpu_blas;
-    rvs_blas *free_gpublas;
-    string msg;
+    uint64_t _matrix_size_a;
```
Comment (manoj-freyr) [inline iet.so/src/iet_worker.cpp:122]: inconsistent naming convention for local vars, see if you can have without starting underscores

File: iet.so/src/iet_worker.cpp [hunk 2]
```diff
@@ -114,73 +114,66 @@ IETWorker::IETWorker():endtest(false) {
...
-        //Print the gflops interval
-        log_interval_gflops(gflops_interval);
-
-        // check end test to avoid unnecessary sleep
-        if (endtest)
-            break;
+        // run GEMM operation
+        if(!gpu_blas->run_blas_gemm()) {
+          endtest = true;
+          break;
```
Comment (manoj-freyr) [inline iet.so/src/iet_worker.cpp:159]: this breaks only from inner loop, is this what you want, if blas fails ? or do we sensibly break out of while loop

Comment (jkottiku) [reply iet.so/src/iet_worker.cpp:159]: Since we are setting endtest=true, it will exit the outer loop too.

**Review by manoj-freyr — APPROVED** [review]
looks good

**Pull request merged**